src/test/java/SpendingAnalyticsTest.java

```diff
@@ -0,0 +1,162 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.common.collect.ImmutableSet;
+import com.google.sps.data.SpendingAnalytics;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class SpendingAnalyticsTest {
+  private static final double ERROR_THRESHOLD = 0.1;
+
+  // Expected values.
+  private static final double WALMART_TOTAL_SPENT = 26.12;
+  private static final double CONTOSO_TOTAL_SPENT = 14.51;
+  private static final double MAIN_STREET_RESTAURANT_SPENT = 29.01;
+
+  // Test Receipt fields.
+  private static final String USER_ID = "1";
+  private static final long TIMESTAMP = 6292020;
+  private static final BlobKey BLOB_KEY = new BlobKey("Test");
+  private static final String IMAGE_URL = "img/walmart-receipt.jpg";
+  private static final double PRICE = 26.12;
+  private static final String STORE = "Walmart";
+  private static final ImmutableSet<String> CATEGORIES =
+      ImmutableSet.of("Cappuccino", "Sandwich", "Lunch");
+  private static final String RAW_TEXT = "Walmart\nAlways Low Prices At Walmart\n";
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void storeAnalyticsWithUniqueStoreNames() {
+    // Only one receipt for each store added:
+    // Walmart: $26.12, Contoso: $14.51, Main Street Restaurant: $29.01
+
+    ImmutableSet<Entity> receipts = TestUtils.createEntities();
```

Comment:
I would probably move the arrangements in `TestUtils.createEntities()` here. We want to ensure that a test can be understood without looking at the definitions of helper methods and not obscure the relation between the arrangement and assertions.